docs/note-new-orders-with-their-fulfillment-locations/script.liquid

```diff
@@ -0,0 +1,72 @@
+{% if event.preview %}
+  {% assign order = hash %}
+  {% assign order["admin_graphql_api_id"] = "gid://shopify/Order/1234567890" %}
+{% endif %}
+
+{% capture query %}
+  query {
+    order(id: {{ order.admin_graphql_api_id | json }}) {
+      fulfillmentOrders(first: 250) {
+        edges {
+          node {
+            assignedLocation {
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+{% endcapture %}
+
+{% assign result = query | shopify %}
+
+{% if event.preview %}
+  {% capture result_json %}
+    {
+      "data": {
+        "order": {
+          "fulfillmentOrders": {
+            "edges": [
+              {
+                "node": {
+                  "assignedLocation": {
+                    "name": "Store #12345"
+                  }
+                }
+              },
+              {
+                "node": {
+                  "assignedLocation": {
+                    "name": "Store #67890"
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = result_json | parse_json %}
+{% endif %}
+
+{% assign fulfillment_location_names = result.data.order.fulfillmentOrders.edges | map: "node" | map: "assignedLocation" | map: "name" | join: ", " %}
+{% assign note = "Fulfilled from: " | append: fulfillment_location_names %}
```

Comment:
There could be an existing order note, so you probably want to make sure and check for it and append this new info afterwards if one exists (after a newline or two).

Reply:
Thanks @tekhaus! A bit of a silly question, but how do I put new lines in the note string? When I do `{% assign note = result.data.order.note | append: "\n\nFulfilled from: " | append: fulfillment_location_names %}` it escapes the backslash and I end up with this: "Fulfilled from: PO Box 21122\n\nFulfilled from: PO Box 21122"

What am I doing wrong here? Thanks!

Reply:
Maybe I should use `capture` instead of `assign`. That seems to work okay.

Reply:
Using `capture` works, but requires some odd white spacing for the result to be clean. Let me know if you think there's a better way.